src/libs/ReportPreviewActionUtils.ts

```diff
@@ -0,0 +1,108 @@
+import type {OnyxCollection} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import CONST from '@src/CONST';
+import type {Policy, Report, TransactionViolation} from '@src/types/onyx';
+import {isApprover as isApprovedMember} from './actions/Policy/Member';
+import {getCurrentUserAccountID} from './actions/Report';
+import {arePaymentsEnabled, getCorrectedAutoReportingFrequency, hasAccountingConnections, isAutoSyncEnabled, isPrefferedExporter} from './PolicyUtils';
+import {
+    getReportTransactions,
+    hasViolations as hasAnyViolations,
+    isClosedReport,
+    isCurrentUserSubmitter,
+    isExpenseReport,
+    isInvoiceReport,
+    isIOUReport,
+    isOpenReport,
+    isPayer,
+    isProcessingReport,
+    isReportApproved,
+    isSettled,
+} from './ReportUtils';
+import {getSession} from './SessionUtils';
+
+function canSubmit(report: Report, violations: OnyxCollection<TransactionViolation[]>, policy?: Policy) {
+    const isExpense = isExpenseReport(report);
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const isOpen = isOpenReport(report);
+    const isManualSubmitEnabled = getCorrectedAutoReportingFrequency(policy) === CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL;
+    const hasViolations = hasAnyViolations(report.reportID, violations);
+    return isExpense && isSubmitter && isOpen && isManualSubmitEnabled && !hasViolations;
+}
+
+function canApprove(report: Report, violations: OnyxCollection<TransactionViolation[]>, policy?: Policy) {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isProcessing = isProcessingReport(report);
+    const isApprovalEnabled = policy ? policy.approvalMode && policy.approvalMode !== CONST.POLICY.APPROVAL_MODE.OPTIONAL : false;
+    const hasViolations = hasAnyViolations(report.reportID, violations);
+    const reportTransactions = getReportTransactions(report?.reportID);
```

Comment:
BUG: View button is shown to admin when employee submits expense. We're getting an empty transactions array here, which makes this condition return false. The issue is gone after a page refresh. It seems like a race condition because we do get the transaction Onyx update

https://github.com/user-attachments/assets/0143c3fa-23bf-4299-9075-a3761b6cbd17